hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.workflows.ingest;
+
+import com.hedera.node.app.spi.key.HederaKey;
+import com.hedera.services.store.models.Account;
+import com.hederahashgraph.api.proto.java.SignatureMap;
+import com.hederahashgraph.api.proto.java.SignedTransaction;
+import com.hederahashgraph.api.proto.java.Transaction;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+/**
+ * Encapsulates the workflow related to transaction ingestion. Given a Transaction, parses,
+ * validates, and submits to the platform.
+ */
+public interface IngestChecker {
+
+    /**
+     * Validates a {@link Transaction}
+     *
+     * @param tx the {@code Transaction}
+     * @throws PreCheckException if validation fails
+     * @throws NullPointerException if {@code tx} is {@code null}
+     */
+    void checkTransaction(@Nonnull Transaction tx) throws PreCheckException;
```

Comment:
I do not know, but I would think so. If we find issues in the transaction, we can skip parsing the signed transaction bytes. But I have not looked into the details of the validation yet, therefore this may not make sense at all.